### PR TITLE
Fix #272: Added delete feature for stories

### LINKED ISSUE #272: [Feature] Add delete feature for a story
## Is your feature request related to a problem? Please describe.

Currently, the stories can only be deleted from the Strapi backend by the admin. This hinders the manager from managing the application properly.

## Describe the solution you'd like

It would be better if there's a delete feature accessible to the Manager only. A delete button should be present in the story page for the manager. If the manager clicks on the delete button, a modal should appear to confirm the action.

## Additional context

NA

### PR DESCRIPTION
**Pull request by kailash360**
**Issue Number**

fixes #272 

**Describe the changes you've made**

I have added the service to get the permission based on the user Id and a service to delete the story based on Id. 

First, the permissions of the current user is fetched using a GraphQL query. If the current user has the right to `Delete Story`, a delete button will be displayed on the story page to delete that story. If the user clicks on the delete button, a pop-up renders to confirm the action from the user. When the story is deleted, the user is redirected to the home page.

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

NA

**Additional context (OPTIONAL)**

NA

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**

https://user-images.githubusercontent.com/75155230/193471701-2a1ffcf1-e9ea-4c5b-9fbb-f342fb5b3cc1.mp4

### CODE REVIEW

**Review by cyntss — CHANGES_REQUESTED** [review]
very small changes requested, the UX and JS look really good otherwise

File: src/pages/Story.js [hunk 1]
```diff
@@ -205,6 +237,14 @@ const Story = (props) => {
                         <Button className='share-story' onClick={copy}>
                           <EOS_CONTENT_COPY className='eos-icons' />
                         </Button>
+                        {permissions.includes('Delete Story') && (
```
Comment (cyntss) [inline src/pages/Story.js:240]: Can you also add some documentation in an .md file stating that a permission with this name "Delete Story" has to be configured on the backend (Strapi) and assigned to a role?

File: cypress/integration/user_account.spec.js [hunk 2]
```diff
@@ -51,13 +51,17 @@ describe("Tests for user's profile page", () => {
   beforeEach(() => {
     Cypress.Cookies.preserveOnce('token')
     cy.restoreLocalStorage()
+
+    cy.wait(5000)
```
[the remark this answers is not in the document]
Comment (kailash360) [reply cypress/integration/user_account.spec.js:55]: Yes, it was added to overcome the timeout

File: cypress/support/commands.js [hunk 3]
```diff
@@ -36,3 +36,9 @@ Cypress.Commands.add('restoreLocalStorage', () => {
     localStorage.setItem(key, LOCAL_STORAGE_MEMORY[key])
   })
 })
+
+Cypress.on('uncaught:exception', (err, runnable) => {
```
[the remark this answers is not in the document]
Comment (kailash360) [reply cypress/support/commands.js:40]: Alright

File: src/pages/Story.js (hunk 1 above)
Comment (kailash360) [reply src/pages/Story.js:240]: Sure

**Comment by kailash360** [comment]
@cyntss 
I have made the changes and pushed them

File: cypress/integration/user_account.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyntss) [reply cypress/integration/user_account.spec.js:55]: let's also remove it as we can see it has no impact. We will merge this branch and test cypress locally

[the remark this answers is not in the document]
Comment (kailash360) [reply cypress/integration/user_account.spec.js]: Done

**Pull request merged**